### DISCUSSION TITLE
Is there any way to have Grid with flex direction "Column"?

### DISCUSSION THREAD

**Discussion by msalafia** [mantinedev/mantine#3325] [Q&A]
Is there any component that should i use to obtain the same behaviour of Grid but in flex-direction="column?"

I would like to have the same behaviour of Grid but in the vertical direction. The Stack component seems not have internal component like a "Stack.Row" or "Stack.Cell" where i can put flexible dimensions like i do with Grid.Col (e.g. span={3}).

**Comment by rtivital** [comment]
Grid with flex-direction column will be just a Stack

**Reply by msalafia** [reply]
Yes, but as far as i know Stack doesn't allow flex partitions of the contaner height as Grid does with Grid.Col.
In other words, there is no "12 Column" partitions in Stack like in Grid.

I have in mind something like:
```
<Stack>
   <Stack.Row span={1}>...<Stack.Row>
   <Stack.Row span={auto}>...<Stack.Row>     //<---- this will take all the remaining space
   <Stack.Row span={1}>...<Stack.Row>
</Stack>
```

**Reply by rtivital** [reply]
> this will take all the remaining space

Remaining space of what?

How it is different from

```ts
<div style={{ display: 'flex' }}>
  <div>{1}</div>
  <div style={{ flex: 1 }}>{2}</div>
  <div>{3}</div>
</div>
```

**Reply by msalafia** [reply]
Wow, interesting. Sorry but i'am not a CSS expert.

Following your example, i imagine i should have to put "flex-direction" column in the outer div, right?

Now, using core layout components of mantine, can i achieve the same result with Stack?

**Reply by rtivital** [reply]
Maybe something like this

```tsx
<Stack>
  <div>{1}</div>
  <div style={{ flex: 1 }}>{2}</div>
  <div>{3}</div>
</Stack>
```

**Reply by msalafia** [reply]
That's ok. I mean, in this way i can just write CSS in the proper element and i get my column and rows. I was just wondering if there was some way to use mantine components and their props (like align, justify, grows, offset, gutter, etc.) to do the job and avoid CSS.
